src/beast/beast/crypto/impl/ripemd_context.h

```diff
@@ -0,0 +1,441 @@
+#define BEAST_RIPEMD_FF(a, b, c, d, e, x, s) {      \
+      (a) += BEAST_RIPEMD_F((b), (c), (d)) + (x);   \
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_GG(a, b, c, d, e, x, s) {      \
+      (a) += BEAST_RIPEMD_G((b), (c), (d)) + (x) + 0x5a827999UL; \
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_HH(a, b, c, d, e, x, s) {      \
+      (a) += BEAST_RIPEMD_H((b), (c), (d)) + (x) + 0x6ed9eba1UL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_II(a, b, c, d, e, x, s) {      \
+      (a) += BEAST_RIPEMD_I((b), (c), (d)) + (x) + 0x8f1bbcdcUL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_JJ(a, b, c, d, e, x, s) {      \
+      (a) += BEAST_RIPEMD_J((b), (c), (d)) + (x) + 0xa953fd4eUL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_FFF(a, b, c, d, e, x, s) {     \
+      (a) += BEAST_RIPEMD_F((b), (c), (d)) + (x);   \
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_GGG(a, b, c, d, e, x, s) {     \
+      (a) += BEAST_RIPEMD_G((b), (c), (d)) + (x) + 0x7a6d76e9UL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_HHH(a, b, c, d, e, x, s) {     \
+      (a) += BEAST_RIPEMD_H((b), (c), (d)) + (x) + 0x6d703ef3UL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_III(a, b, c, d, e, x, s) {     \
+      (a) += BEAST_RIPEMD_I((b), (c), (d)) + (x) + 0x5c4dd124UL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10);  }
+#define BEAST_RIPEMD_JJJ(a, b, c, d, e, x, s) {     \
+      (a) += BEAST_RIPEMD_J((b), (c), (d)) + (x) + 0x50a28be6UL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+
+template <class = void>
+void ripemd_load (std::array<std::uint32_t, 16>& X,
+    void const* block)
+{
+    std::uint8_t const* p =
+        reinterpret_cast<
+            std::uint8_t const*>(block);
+    for(int i = 0; i < 16; ++i)
+    {
+        X[i] =
+            ((std::uint32_t) *((p)+3) << 24) |
+            ((std::uint32_t) *((p)+2) << 16) |
+            ((std::uint32_t) *((p)+1) <<  8) |
+            ((std::uint32_t) * (p));
+        p += 4;
+    }
+}
+
+template <class = void>
+void ripemd_compress (ripemd160_context& ctx,
+        std::array<std::uint32_t, 16>& X) noexcept
+{
+    std::uint32_t aa  = ctx.h[0];
+    std::uint32_t bb  = ctx.h[1];
+    std::uint32_t cc  = ctx.h[2];
+    std::uint32_t dd  = ctx.h[3];
+    std::uint32_t ee  = ctx.h[4];
+    std::uint32_t aaa = ctx.h[0];
```

Comment:
There's definitely more to be gained by disturbing the original code as little as possible